### PR TITLE
Lep 57 Part II - DRY up controller code

### PR DESCRIPTION
**Pull request by starswan**
https://dsdmoj.atlassian.net/browse/LEP-57

There is a lot of duplication in the controllers - this extracts and removes the duplication, making the actual change for LEP-57 much more straightforward

### CODE REVIEW

File: app/services/creators/proceeding_types_creator.rb [hunk 1]
```diff
@@ -1,44 +1,34 @@
...
-      self.proceeding_types = assessment.proceeding_types.create!(proceeding_types_attributes)
-    rescue StandardError => e
-      raise CreationError, "#{e.class} - #{e.message}"
-    end
+      def create_records(assessment:, proceeding_types_params:)
+        create_proceeding_types(assessment:, proceeding_types_params:)
+        Result.new(errors: []).freeze
+      # rescue ActiveRecord::RecordInvalid => e
+      #   Result.new(errors: e.record.errors.full_messages).freeze
+      rescue ActiveRecord::RecordNotUnique => e
```
Comment (ripan) [inline app/services/creators/proceeding_types_creator.rb:21]: @starswan What do you think of rescuing ActiveRecord::RecordInvalid ?

File: app/services/creators/regular_transactions_creator.rb [hunk 2]
```diff
@@ -1,58 +1,34 @@
...
-    def create_records
-      ActiveRecord::Base.transaction do
-        assessment
-        create_regular_transaction
-      rescue CreationError => e
-        self.errors = e.errors
+    class << self
+      def call(regular_transaction_params:, gross_income_summary:)
+        ActiveRecord::Base.transaction do
+          create_regular_transaction(regular_transactions: regular_transaction_params.fetch(:regular_transactions, []),
```
Comment (ripan) [inline app/services/creators/regular_transactions_creator.rb:12]: @starswan Do we need to validate json before creation?

File: app/controllers/cash_transactions_controller.rb [hunk 3]
```diff
@@ -1,21 +1,15 @@
-class CashTransactionsController < ApplicationController
+class CashTransactionsController < CreationController
   def create
-    if creation_service.success?
-      render_success
-    else
-      render_unprocessable(creation_service.errors)
-    end
+    json_validate_and_render("cash_transaction", cash_transaction_params, lambda {
```
[the remark this answers is not in the document]
Comment (starswan) [reply app/controllers/cash_transactions_controller.rb:3]: When the swagger work comes in, the schema name becomes pluralized. This PR is just a refactor

File: app/controllers/assessments_controller.rb [hunk 4]
```diff
@@ -25,6 +22,10 @@ def show
 
 private
 
+  def validate
+    validate_json_schema "assessment", assessment_params
```
[the remark this answers is not in the document]
Comment (starswan) [reply app/controllers/assessments_controller.rb:26]: same as above

File: app/services/creators/full_assessment_creator.rb [hunk 5]
```diff
@@ -22,16 +22,16 @@ def call(remote_ip:, params:)
 
       CREATE_FUNCTIONS = [
         lambda { |assessment, params|
-          Creators::ProceedingTypesCreator.call(assessment_id: assessment.id,
+          Creators::ProceedingTypesCreator.call(assessment:,
                                                 proceeding_types_params: { proceeding_types: params[:proceeding_types] })
         },
         lambda { |assessment, params|
-          Creators::ApplicantCreator.call(assessment_id: assessment.id,
+          Creators::ApplicantCreator.call(assessment:,
```
[the remark this answers is not in the document]
Comment (ripan) [reply app/services/creators/full_assessment_creator.rb:29]: sorry, ignore my comment. LGTM 👍

File: app/services/creators/proceeding_types_creator.rb (hunk 1 above)
Comment (starswan) [reply app/services/creators/proceeding_types_creator.rb]: I've added a validation so it now raises RecordInvalid

**Review by ripan — APPROVED** [review]

**Pull request merged**